src/Factory/DelegatorFactoryInterface.php

```diff
@@ -23,6 +23,7 @@ interface DelegatorFactoryInterface
      * A factory that creates delegates of a given service
      *
      * @param  string             $name
+     * @psalm-param callable():mixed $callback
      * @param  null|array         $options
      * @return object
```

Comment:
Since `ContainerInterface#get` can return `mixed`, this actually has to be `mixed` :-(

But yes, having a templated type would help for sure.

Reply:
Yeah, but the commented line is about `@return object` (wrong)

Reply:
Not sure if I can follow.

Do you want me to change existing documentation that we end up having bc breaks? as of now, due to the lack of typing, callable factories can return mixed. When I enforce `object` now, this will be pain probably. 

Or do you want me to open up the return value of the delegator to be mixed?

Reply:
I think I am fine with both as I don't have that much experience with adding psalm types on such a widely used component. As always, I trust your experience. :-)